pandas/tests/io/sas/test_sas7bdat.py

```diff
@@ -217,6 +217,14 @@ def test_zero_variables(datapath):
         pd.read_sas(fname)
 
 
+def test_corrupt_read(datapath):
+    # We don't really care about the exact failure, the important thing is
+    # that the resource should be cleaned up afterwards (BUG #35566)
+    fname = datapath("io", "sas", "data", "corrupt.sas7bdat")
```

Comment:
Can you test that the file is actually closed?

Reply:
If you have any ideas how to test for that I am all ears!
Certain builds will emit a ResourceWarning but it is only for information - I have visually monitored this, and this patch adds a new warning with the test case and removes it with the change to sasreader.py
(It looks like ResourceWarnings are periodically manually cleaned up, eg #13292, #23680, ..., but there is no mechanism in place for systematically catching them.)

Reply:
yeah this is fine; we don't really deal well with ResourceWarnings; happy to have a patch which elevates these to fail the build (separate PR)